src/chat/entities.rs

```diff
@@ -174,6 +174,9 @@ pub struct ChatMessage {
     /// User ID of the sender. Maybe be `None` for `Event` message types, possibly others
     pub sender_id: Option<i64>,
 
+    /// Time that the message was sent
+    pub send_time: u32,
```

Comment:
Looks like `datetime_utc_ts_seconds_from_any` doesn't support Serde::Serialize, and none of the other locations DateTime's are being deserialized aren't including Serialize in what they're deriving

I'll poke around to see if there's an obvious answer, but I think this'll need a specific format to serialize from so that it can be serialized too as well.

Reply:
Ah, yeah that's unfortunate. I suppose it's because it wouldn't be able to decide if it should be serialised as a number or string.

Maybe we could use [TimestampSeconds<i64>](https://docs.rs/serde_with/1.14.0/serde_with/struct.TimestampSeconds.html) assuming the chat definitely uses numbers and not strings - the trove examples for the http api seem to have a mix, not sure if the examples match the actual api behaviour there.

Reply:
I've got that TimestampSeconds path working with my test usage with expecting an i64 against my test code.